src/org/labkey/test/tests/component/QueryGridTest.java

```diff
@@ -0,0 +1,443 @@
+                    Map.of("Name", sampleSetDataGenerator.randomString(25),
+                            "Description", "used to test issue 39011",
+                            "intColumn", i,
+                            "stringColumn", "filtered_x" + sampleSetDataGenerator.randomString(20)));
+        }
+        sampleSetDataGenerator.insertRows();
+
+        CoreComponentsTestPage testPage = CoreComponentsTestPage.beginAt(this, getProjectName());
+        QueryGrid grid = testPage.getQueryGrid("samples", "filtered_grid_multipageselection_samples");
+
+        // confirm that selectAll is limited to just the records in a filtered set
+        grid.filterOn("Description", "=", "used to test issue 39011");
+        grid.selectAllRows();
+        grid.clearSortsAndFilters();
+        assertThat(grid.getSelectionStatusCount(), is("64 of 124 selected"));
+
+        grid.clearAllSelections();
+        assertThat(grid.getSelectedRows().size(), is(0));
+
+        // confirm that search behaves the same way as a filter does
+        grid.search("filtered_x");
+        grid.selectAllRows();
+        grid.clearSortsAndFilters();
+        assertThat(grid.getSelectionStatusCount(), is("64 of 124 selected"));
+
+        // clean up domain after on success
+        sampleSetDataGenerator.deleteDomain(createDefaultConnection());
+    }
+
+    @Test
+    public void testFilterOnColumnNotInView() throws Exception
+    {
+        // create a sampleType domain to use in this case
+        SampleTypeDefinition props = new SampleTypeDefinition("search_for_description_expression").setFields(standardTestSampleFields());
+        TestDataGenerator sampleSetDataGenerator = SampleTypeAPIHelper.createEmptySampleType(getProjectName(), props);
+        sampleSetDataGenerator.generateRows(30);
+        for (int i=0; i < 5; i++)  // add some rows with a description we can filter on
+        {
+            sampleSetDataGenerator.addCustomRow(
+                    Map.of("Name", sampleSetDataGenerator.randomString(25),
+                            "Description", "used to search for off-view data",
+                            "intColumn", i,
+                            "stringColumn", "filtered_x" + sampleSetDataGenerator.randomString(20)));
+        }
+        sampleSetDataGenerator.insertRows();
+
+        CoreComponentsTestPage testPage = CoreComponentsTestPage.beginAt(this, getProjectName());
+        QueryGrid grid = testPage.getQueryGrid("samples", "search_for_description_expression");
+
+        // prove we can filter on this criteria
+        grid.filterOn("Description", "contains", "off-view");
+        assertThat(grid.getGridBar().pager().summary(), is ("1 - 5"));
+        grid.clearSortsAndFilters();
+
+        sampleSetDataGenerator.deleteDomain(createDefaultConnection());
+    }
+
+    @Ignore // until https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=41169 is resolved
+    @Test
+    public void testEmptyNotEmptyFilter() throws Exception
```

Comment:
Why `@Ignore` this test? It looks like the "has any value" check will pass with the current (incorrect) behavior which seems like the right approach.

Reply:
If it was behaving correctly, it would say 1- 20 of 30 but it instead current behavior has it saying 1-20 of 35 because the custom rows (which don't have values in Int Column) aren't filtered out.  
@Ignore is there so it's not noisy until the bug is addressed.  If that's not right, I'm happy to change it